pkgs/unified_analytics/lib/unified_analytics.dart

```diff
@@ -2,6 +2,7 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-export 'src/analytics.dart' show Analytics;
+export 'src/analytics.dart' show Analytics, NoOpAnalytics;
+export 'src/config_handler.dart' show ToolInfo;
```

Comment:
any type implemented in this package that can be returned from a public method exported from this package should also be exported.

Reply:
Just curious, but is this a Flutter rule? Because I think it's perfectly reasonable for a package to export a public interface and to have one or more private implementations of that interface that aren't exported but can be returned, and that's how I'd view `NoOpAnalytics`. But I'm probably missing something.

Reply:
Sorry for the noise. Just saw the issue update, which explains why you're exporting it.

Reply:
Well, maybe not a rule, but what's the advantage of keeping the interface unexported (especially if we're making instances of it publicly available to callers)?

This was awkward when I tried to implement `NoOpAnalytics` internally, where I had to do:

```dart
  @override
  final parsedTools = {};
```

Although maybe this only feels wrong because for Flutter open source we require explicit types?

Reply:
Thank you for the catch, Chris! And Brian, I think he's referring to the public getter `parsedTools` under the `Analytics`.

```dart
  /// Returns a map object with all of the tools that have been parsed
  /// out of the configuration file
  Map<String, ToolInfo> get parsedTools;
```

You need access to the `ToolInfo` class to implement this class

Reply:
> Although maybe this only feels wrong because for Flutter open source we require explicit types?

Ah, I didn't realize this was a flutter rule, I thought explicit types were the default for Dart

Reply:
Effective dart recommends relying on type inference: https://dart.dev/guides/language/effective-dart/usage#do-follow-a-consistent-rule-for-var-and-final-on-local-variables

The rationale for requiring explicit types in the Flutter SDK is: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-var-and-dynamic